opm/autodiff/GraphColoring.hpp

```diff
@@ -0,0 +1,244 @@
+/*
+  Copyright 2018 Equinor
+
+  This file is part of the Open Porous Media project (OPM).
+
+  OPM is free software: you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation, either version 3 of the License, or
+  (at your option) any later version.
+
+  OPM is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with OPM.  If not, see <http://www.gnu.org/licenses/>.
+*/
+#ifndef OPM_GRAPHCOLORING_HEADER_INCLUDED
+#define OPM_GRAPHCOLORING_HEADER_INCLUDED
+
+#include <vector>
+#include <deque>
+#include <tuple>
+#include <algorithm>
+#include <numeric>
+#include <queue>
+
+namespace Opm
+{
+
+namespace Detail
+{
+template<class Graph>
+std::size_t colorGraphWelshPowell(const Graph& graph,
```

Comment:
If I'm being **extremely** pedantic you need to [include](https://en.cppreference.com/w/cpp/types/size_t) one of `<cstddef>`, `<cstdio>`, `<cstdlib>`, `<cstring>`, `<ctime>`, or `<cwchar>` in order to use `std::size_t`.  It is of course likely (almost guaranteed) that `<vector>` will do that transitively.